frontend/appflowy_flutter/lib/plugins/database/widgets/cell_editor/relation_cell_editor.dart

```diff
@@ -14,126 +16,252 @@ import '../../application/cell/bloc/relation_row_search_bloc.dart';
 class RelationCellEditor extends StatelessWidget {
   const RelationCellEditor({
     super.key,
-    required this.selectedRowIds,
   });
 
-  final List<String> selectedRowIds;
-
   @override
   Widget build(BuildContext context) {
     return BlocBuilder<RelationCellBloc, RelationCellState>(
       builder: (context, cellState) {
-        if (cellState.relatedDatabaseMeta == null) {
-          return const _RelationCellEditorDatabaseList();
+        return cellState.relatedDatabaseMeta == null
+            ? const _RelationCellEditorDatabaseList()
+            : _RelationCellEditorRowList(
+                relatedDatabaseMeta: cellState.relatedDatabaseMeta!,
+                selectedRowIds: cellState.rows.map((e) => e.rowId).toList(),
+              );
+      },
+    );
+  }
+}
+
+class _RelationCellEditorRowList extends StatefulWidget {
+  const _RelationCellEditorRowList({
+    required this.relatedDatabaseMeta,
+    required this.selectedRowIds,
+  });
+
+  final DatabaseMeta relatedDatabaseMeta;
+  final List<String> selectedRowIds;
+
+  @override
+  State<_RelationCellEditorRowList> createState() =>
+      _RelationCellEditorRowListState();
+}
+
+class _RelationCellEditorRowListState
+    extends State<_RelationCellEditorRowList> {
+  final textEditingController = TextEditingController();
+  late final FocusNode focusNode;
+  late final bloc = RelationRowSearchBloc(
+    databaseId: widget.relatedDatabaseMeta.databaseId,
+  );
+
+  @override
+  void initState() {
+    super.initState();
+    focusNode = FocusNode(
+      onKeyEvent: (node, event) {
```

Comment:
I feel like this is overengineered..

IIRC ListView offers this type of focus traversal out of the box.. Just needs to have focus first.. (Eg. you could just build the arrow down key event from the search field, and omit the rest?)

Reply:
I'm not sure how to do that without having the text field lose focus after arrow up or down. Can you show me an example?

Reply:
I see what you mean, but that's non-standard behavior no? Is that really what we want to do?

And if so, I'd rather see a common implementation that we can reuse if that's the case.

Reply:
What is standard behavior, do you mean the focus traversal that you'd get with tab and shift tab? If so, it's very common to deviate from this when catering to arrow-key-focused UX.

I'd be down for an abstraction at the widget level if it's not going to make the code more complex. We would only have 2 implementations so far including this one. If that number keeps going up, the abstraction would definitely be helpful.